airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -182,8 +183,14 @@ def format_url(host: str) -> str:
 
         return host
 
-    def _render_log_id(self, ti: TaskInstance, try_number: int) -> str:
+    def _render_log_id(self, ti: TaskInstance | TaskInstanceKey, try_number: int) -> str:
+        from airflow.models.taskinstance import TaskInstance, TaskInstanceKey
+
         with create_session() as session:
+            if isinstance(ti, TaskInstanceKey):
+                ti = TaskInstance.get_by_key(ti, session)
```

Comment:
also this is stale code which was based on one of the earlier commits in the base PR for TCL.
I guess we will have to make the method [ensure_ti](https://github.com/apache/airflow/blob/main/airflow/utils/log/file_task_handler.py#L135) public and re-use here or something like that? @dstandish

Reply:
right yeah i will push code

for now i'll just duplicate the private method

want to think it through before adding to public interface